tests/Unit/HC/amdgcn_ds_bpermute.cpp

```diff
@@ -35,7 +35,8 @@ bool run_test(const int num) {
   T max = std::is_signed<T>::value ? INT_MAX : UINT_MAX;
   T min = std::is_signed<T>::value ? INT_MIN : 0;
 
-  std::uniform_int_distribution<T> distribution(min, max);
+  typedef typename std::conditional<std::is_signed<T>::value, int, unsigned int>::type _RangeType;
+  std::uniform_int_distribution<_RangeType> distribution(static_cast<_RangeType>(min), static_cast<_RangeType>(max));
```

Comment:
libstdc++'s uniform_int_distribution enforces the input template type to be integers.  This change is to allow the unit test template to work with floats